### PR TITLE
Add regression tests on optimise

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by OlivierBlanvillain** [comment]
Maybe we should create an issue to not forget about these pattern matching warnings?

File: compiler/test/dotty/tools/dotc/CompilationTests.scala [hunk 1]
```diff
@@ -186,6 +189,7 @@ class CompilationTests extends ParallelTesting {
 
   @Test def runAll: Unit = {
     compileFilesInDir("../tests/run", defaultOptions) +
+    compileFilesInDir("../tests/run", defaultOptimised) +
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/dotc/CompilationTests.scala]: If you run everything in `tests/run` with `defaultOptimised`, aren't the lines below redundant?

Comment (nicolasstucki) [reply compiler/test/dotty/tools/dotc/CompilationTests.scala]: Yes, I will remove them. Thanks

**Comment by allanrenucci** [comment]
I think this PR is the cause of the spurious failures than can be observed [here](http://dotty-ci.epfl.ch/lampepfl/dotty/363/6) and [here](http://dotty-ci.epfl.ch/lampepfl/dotty/364/5). I think we should revert it

**Comment by nicolasstucki** [comment]
Then revert it, I will test again next week

**Pull request merged**